maven-plugin/src/main/java/org/sonatype/nexus/maven/staging/StagingDeployMojo.java

```diff
@@ -230,33 +230,34 @@ private void attachToIndex(
       final ArtifactRepository artifactRepository) throws IOException
   {
 
-    Optional<String> maybePluginPrefix = artifact.getMetadataList()
+    String pluginPrefix = artifact.getMetadataList()
         .stream()
-        .filter(metadata -> metadata instanceof GroupRepositoryMetadata)
+        .filter(GroupRepositoryMetadata.class::isInstance)
         .map(metadata -> ((GroupRepositoryMetadata) metadata).getMetadata().getPlugins())
         .filter(plugins -> !plugins.isEmpty())
         .flatMap(List::stream)
         .map(Plugin::getPrefix)
-        .findFirst();
+        .findFirst()
+        .orElse(null);
 
-    Optional<String> maybePomFileName = artifact.getMetadataList()
+    String pomFileName = artifact.getMetadataList()
         .stream()
-        .filter(metadata -> metadata instanceof ProjectArtifactMetadata)
+        .filter(ProjectArtifactMetadata.class::isInstance)
         .map(projectMetadata -> projectMetadata.getLocalFilename(artifactRepository))
-        .findFirst();
-
-    Optional<String> maybeClassifier = Optional.ofNullable(artifact.getClassifier());
+        .findFirst()
+        .orElse(null);
 
     ArtifactInfo artifactInfo = new ArtifactInfo();
+    artifactInfo.setArtifactPath(artifactRepository.pathOf(artifact));
     artifactInfo.setGroup(artifact.getGroupId());
     artifactInfo.setArtifactId(artifact.getArtifactId());
     artifactInfo.setVersion(artifact.getVersion());
     artifactInfo.setTag(tag);
-    artifactInfo.setClassifier(maybeClassifier.orElse(NOT_APPLIES));
+    artifactInfo.setClassifier(artifact.getClassifier());
```

Comment:
So before it was passing in "n/a" and returning "n/a" but now it's passing null and returning null? Seems like either the n/a logic isn't necessary anymore inside ArtifactInfo or the getter should be returning the stored "n/a" and not converting to back to null.

Reply:
As mentioned previously, n/a isn't really needed by the implementation, but may be intended/beneficial (for human readable) when serialised within the JSON in the .index file.

Reply:
The old version of the plugin used n/a only because the index was properties file with values delimited by `:` on each line. No reason not to use null here.

Reply:
Changed to null in [ead71ff](https://github.com/sonatype/nxrm3-maven-plugin/pull/36/commits/ead71ffee2f92e843ac341fe6d8de9bf3cbce424)